hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoTransferTransitionLogic.java

```diff
@@ -63,14 +70,19 @@ public CryptoTransferTransitionLogic(
 			GlobalDynamicProperties dynamicProperties,
 			ImpliedTransfersMarshal impliedTransfersMarshal,
 			PureTransferSemanticChecks transferSemanticChecks,
-			ExpandHandleSpanMapAccessor spanMapAccessor
-	) {
+			ExpandHandleSpanMapAccessor spanMapAccessor,
+			OptionValidator validator,
+			SideEffectsTracker sideEffectsTracker) {
 		this.txnCtx = txnCtx;
 		this.ledger = ledger;
 		this.spanMapAccessor = spanMapAccessor;
 		this.dynamicProperties = dynamicProperties;
 		this.transferSemanticChecks = transferSemanticChecks;
 		this.impliedTransfersMarshal = impliedTransfersMarshal;
+		this.validator = validator;
+		this.sideEffectsTracker = sideEffectsTracker;
+		this.scopedCheck = new MerkleAccountScopedCheck(dynamicProperties, validator);
+		this.transferLogic = new TransferLogic(scopedCheck, dynamicProperties, validator, ledger.getAccountsLedger(), ledger.getTokenLedger(), ledger.getNftsLedger(), ledger.getTokenRelsLedger(), sideEffectsTracker);
```

Comment:
I think that instead of using `ledger.get...` we would need to provide instances of the ledger. Ideally we won't need the ledger at all

Reply:
I think this is the reason I haven't made TransferLogic get the whole HederaLedger, needed to make the current context fit the idea we are trying to achieve.